src/native/corehost/hostpolicy/deps_format.cpp

```diff
@@ -180,31 +180,153 @@ void deps_json_t::reconcile_libraries_with_targets(
     }
 }
 
-// Returns the RID determined (computed or fallback) for the platform the host is running on.
-pal::string_t deps_json_t::get_current_rid(const rid_fallback_graph_t& rid_fallback_graph)
+namespace
 {
-    pal::string_t currentRid = get_current_runtime_id(false /*use_fallback*/);
+    #define CURRENT_ARCH_SUFFIX _X("-") _STRINGIFY(CURRENT_ARCH_NAME)
+    #define RID_CURRENT_ARCH_LIST(os) \
+        _X(os) CURRENT_ARCH_SUFFIX, \
+        _X(os),
+
+    const pal::char_t* s_host_rids[] =
+    {
+#if defined(TARGET_WINDOWS)
+        RID_CURRENT_ARCH_LIST("win")
+#elif defined(TARGET_OSX)
+        RID_CURRENT_ARCH_LIST("osx")
+        RID_CURRENT_ARCH_LIST("unix")
+#elif defined(TARGET_LINUX_MUSL)
+        RID_CURRENT_ARCH_LIST("linux-musl")
+        RID_CURRENT_ARCH_LIST("linux")
+        RID_CURRENT_ARCH_LIST("unix")
+#elif defined(TARGET_ANDROID)
+        RID_CURRENT_ARCH_LIST("linux-bionic")
+        RID_CURRENT_ARCH_LIST("linux")
+        RID_CURRENT_ARCH_LIST("unix")
+#else
+        RID_CURRENT_ARCH_LIST(FALLBACK_HOST_OS)
```

Comment:
```suggestion
        RID_CURRENT_ARCH_LIST(CURRENT_OS_NAME)
```

Reply:
For source-built .NET, we want to ensure the build target rid gets probed.

Isn't this omitting that rid?

Reply:
Yes, combining them together, I think we need something like:
```suggestion
        RID_CURRENT_ARCH_LIST(FALLBACK_HOST_OS) # covers non-portable RIDs
        RID_CURRENT_ARCH_LIST(CURRENT_OS_NAME) # covers "linux" and non-linux like "freebsd". "illumos"
        RID_CURRENT_ARCH_LIST("unix")
```